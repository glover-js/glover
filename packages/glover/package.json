{
  "name": "glover",
  "description": "TBD",
  "version": "1.0.0",
  "main": "dist/index.cjs.js",
  "module": "dist/index.esm.js",
  "bin": {
    "glover": "dist/bin/cli.js"
  },
  "dependencies": {
    "glover-jest": "1.0.0",
    "glover-logger": "1.0.0",
    "glover-resolver": "1.0.0",
    "inquirer": "^6.2.1",
    "mustache": "^3.0.1",
    "ora": "^3.0.0",
    "yargs": "^12.0.5"
  },
  "repository": {
    "type": "git",
    "url": "https://github.com/glover-js/glover"
  },
  "bugs": {
    "url": "https://github.com/glover-js/glover/issues"
  },
  "homepage": "https://glover-js.github.io/",
  "license": "MIT",
  "keywords": [
    "TBD"
  ]
}
