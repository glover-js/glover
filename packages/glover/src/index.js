// @flow
import type { RunConfig } from 'types/Config';
import type { RunResult } from 'types/Result';

import brew from './brew';
import pick from './pick';
import compose from './compose';
import CommandNotFoundError from './utils/CommandNotFoundError';

async function run(config: RunConfig): Promise<RunResult> {
  if (config.command === 'brew') {
    return brew(config);
  }
  if (config.command === 'pick') {
    return pick(config);
  }
  if (config.command === 'compose') {
    return compose(config);
  }
  throw new CommandNotFoundError(config.command);
}

export default {
  run,
  brew,
  pick,
  compose,
};
