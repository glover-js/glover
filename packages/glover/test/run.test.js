import glover from '../src';

describe('Tests for glover.run', () => {
  it('runs brew', async () => {
    const runConfig = { command: 'brew' };
    await expect(glover.run(runConfig)).resolves.toEqual({ command: 'brew' });
  });
  it('runs pick', async () => {
    const runConfig = { command: 'pick' };
    await expect(glover.run(runConfig)).resolves.toEqual({ command: 'pick' });
  });
  it('runs compose', async () => {
    const runConfig = { command: 'compose' };
    await expect(glover.run(runConfig)).resolves.toEqual({
      command: 'compose',
    });
  });
  it('rejects unknown commands', async () => {
    const runConfig = {
      command: 'unknown',
      rawArgv: ['', '', 'unknown', 'argument'],
    };
    await expect(glover.run(runConfig)).rejects.toThrowErrorMatchingSnapshot();
  });
});
