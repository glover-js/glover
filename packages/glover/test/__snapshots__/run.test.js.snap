// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Tests for glover.run rejects unknown commands 1`] = `"Error: unknown is not a valid glover command"`;
